### PR TITLE
Fix issue in collecting cases for sharing deployments

### PR DESCRIPTION
**Pull request by lizzha**

### CODE REVIEW

File: TestControllers/TestController.psm1 [hunk 1]
```diff
@@ -245,24 +245,9 @@ Class TestController
...
-				}
-				if ($test.AdditionalHWConfig.SwitchName) {
-					$switchName = $test.AdditionalHWConfig.SwitchName
-				}
-			}
-
-			# Add case to hashtable
 			if ($test.setupType) {
-				$key = "$($test.setupType),$($test.OverrideVMSize),$networking,$diskType,$osDiskType,$switchName"
+				$key = "$($test.setupType),$($test.OverrideVMSize),$($test.AdditionalHWConfig.Networking),$($test.AdditionalHWConfig.DiskType)," +
```
[the remark this answers is not in the document]
Comment (juhlee-microsoft) [reply TestControllers/TestController.psm1]: OK, if you post any test using AdditionalHWConfig, I will sign off and merge it.

**Comment by lizzha** [comment]
The change is in collecting cases. Here're the outputs of collecting cases for network performance test
Command:
.\Run-LisaV2.ps1 -TestPlatform "Azure" -RGIdentifier liz-test -ARMImageName "Canonical UbuntuServer 16.04-LTS Latest"  -XMLSecretFile D:\Documents\Github\lizzha\AzureCredential.xml -TestLocation eastus -TestCategory Performance -TestArea Network

01/17/2019 06:44:14 : [INFO ] Validating XML Files from D:\LISAv2\AV20 folder recursively...
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-SRIOV
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-NETPERF-MAX-PPS-Synthetic
01/17/2019 06:44:14 : [INFO ] Collected: PERF-NETWORK-TCP-NETPERF-MAX-PPS-SRIOV
01/17/2019 06:44:14 : [INFO ] 14 Test Cases have been collected
01/17/2019 06:44:14 : [INFO ] NTTTCP_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] NTTTCP_TCP_CONNECTIONS=(1 2 4 8 16 32 64 128 256 512 1024 2048 4096 6144 8192 10240) injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] NTTTCP_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] NTTTCP_TCP_CONNECTIONS=(1 2 4 8 16 32 64 128 256 512 1024 2048 4096 6144 8192 10240) injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] NTTTCP_VERSION=v1.3.4 injected to case PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] NTTTCP_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] NTTTCP_UDP_CONNECTIONS=(1 2 4 8 16 32 64 128 256 512 1024) injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
01/17/2019 06:44:14 : [INFO ] NTTTCP_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] NTTTCP_UDP_CONNECTIONS=(1 2 4 8 16 32 64 128 256 512 1024) injected to case PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-Synthetic
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-SRIOV
01/17/2019 06:44:14 : [INFO ] IPERF3_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-Synthetic
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-Synthetic
01/17/2019 06:44:14 : [INFO ] IPERF3_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-SRIOV
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-SRIOV
01/17/2019 06:44:14 : [INFO ] IPERF3_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC
01/17/2019 06:44:14 : [INFO ] PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC-BUFFER-LENGTHS=(32 64 128 256 512 1024 2048 4096 8192 16384 32768 65536) injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC
01/17/2019 06:44:14 : [INFO ] IPERF3_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV
01/17/2019 06:44:14 : [INFO ] PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV-BUFFER-LENGTHS=(32 64 128 256 512 1024 2048 4096 8192 16384 32768 65536) injected to case PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV
01/17/2019 06:44:14 : [INFO ] NETPERF_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-Synthetic
01/17/2019 06:44:14 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-Synthetic
01/17/2019 06:44:15 : [INFO ] NETPERF_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-SRIOV
01/17/2019 06:44:15 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-SRIOV
01/17/2019 06:44:15 : [INFO ] NETPERF_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-NETPERF-MAX-PPS-Synthetic
01/17/2019 06:44:15 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-NETPERF-MAX-PPS-Synthetic
01/17/2019 06:44:15 : [INFO ] NETPERF_RUNTIME_IN_SECONDS=300 injected to case PERF-NETWORK-TCP-NETPERF-MAX-PPS-SRIOV
01/17/2019 06:44:15 : [INFO ] HYPERV_PERF_NIC_NAME=SRIOV injected to case PERF-NETWORK-TCP-NETPERF-MAX-PPS-SRIOV
01/17/2019 06:44:15 : [INFO ] Starting the test
01/17/2019 06:44:15 : [INFO ] Current test setup: M1S1
01/17/2019 06:44:15 : [INFO ] Checking the subscription usage...

**Comment by lizzha** [comment]
For full test logs, let me run two cases with and without AddtionalHWConfig, and post logs here.

**Comment by lizzha** [comment]
@juhlee-microsoft test pass against Ubuntu 18.04
[LISAv2 Test Results Summary]
Test Run On           : 01/17/2019 06:48:42
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : Latest
Total Test Cases      : 2 (2 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:24

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-ROOT-DEVICE-TIMEOUT-CHECK                                                     PASS                 3.09
    2 BVT-VERIFY-DEPLOYMENT-PROVISION-EPHEMERAL-MANAGED-DISK                            PASS                  4.4
        FirstBoot : PASS
        FirstBoot : Call Trace Verification : PASS
        Reboot : PASS
        Reboot : Call Trace Verification : PASS

**Pull request merged**